### PR TITLE
DO NOT MERGE: MCOL-4314 Shared storage manager cluster initialization bug fix

### PR DESCRIPTION
**Pull request by jmrojas2332**

### CODE REVIEW

File: oam/install_scripts/mcs-loadbrm.py.in [hunk 1]
```diff
@@ -120,9 +120,6 @@ node.".format(e))
                     if storage.lower() == 's3' and bucket.lower() != 'some_bucket':
                         dbrmroot = BYPASS_SM_PATH
 
-                        if not os.path.exists(dbrmroot):
-                            os.makedirs(dbrmroot)
```
Comment (jmrojas2332) [inline oam/install_scripts/mcs-loadbrm.py.in:124]: Creating a directory, which is creating a loadbrm failure when attempted to load this directory as brm metadata

File: oam/install_scripts/columnstore-post-install.in [hunk 2]
```diff
@@ -310,16 +310,13 @@ chmod 777 /dev/shm
...
 
-    if [ ! -e $IFLAG ]; then
-       touch $IFLAG
-       echo "Populating the engine initial system catalog."
-       systemctl start mariadb-columnstore
-    fi
+    echo "Populating the engine initial system catalog."
+
+    # prevent nodes using shared storage manager from stepping on each other when initializing
+    flock /var/lib/columnstore/storagemanager -c "systemctl start mariadb-columnstore"
```
Comment (drrtuy) [inline oam/install_scripts/columnstore-post-install.in:319]: I wonder if flock must defend against multiple runs of the command on the same node?

File: oam/install_scripts/mcs-loadbrm.py.in (hunk 1 above)
Comment (drrtuy) [reply oam/install_scripts/mcs-loadbrm.py.in:124]: Does it mean that we have to create directories beforehand ?

File: oam/install_scripts/columnstore-post-install.in [hunk 3]
```diff
@@ -310,16 +310,14 @@ chmod 777 /dev/shm
...
+    systemctl stop mariadb-columnstore # shutdown after initialization
```
Comment (jmrojas2332) [inline oam/install_scripts/columnstore-post-install.in:320]: This line `systemctl stop mariadb-columnstore` does three things. It keeps columnstore consistent on both docker and systemd environments, it keeps columnstore consistent with mariadb status after install, but most importantly it allows user to configure storagemanager.cnf after installation without hitting error on savebrm when columnstore is reloaded after modifying config (i.e., loaded localstorage, but attempted to savebrm without a storagemanager process.

Comment (jmrojas2332) [reply oam/install_scripts/columnstore-post-install.in:320]: Verified our official docs have user starting up mariadb-columnstore via systemctl start mariadb-columnstore so this does not cause any issues on user side

File: oam/install_scripts/columnstore-post-install.in (hunk 2 above)
Comment (jmrojas2332) [reply oam/install_scripts/columnstore-post-install.in:319]: without any arguments, flock defaults to be an exclusive lock (write lock) so I believe this is protected.

Comment (drrtuy) [reply oam/install_scripts/columnstore-post-install.in:319]: It was just a news to me that /var/lib/columnstore/storagemanager is shared across nodes.

File: oam/install_scripts/columnstore-post-install.in (hunk 3 above)
Comment (drrtuy) [reply oam/install_scripts/columnstore-post-install.in:320]: Could you share the docs that you refer to? I'm not aware about installation guide for community version.

File: oam/install_scripts/columnstore-post-install.in (hunk 2 above)
Comment (jmrojas2332) [reply oam/install_scripts/columnstore-post-install.in:319]: https://mdbcdt.com/DOCS-2085/deploy/community-single-columnstore/#post-installation

File: oam/install_scripts/columnstore-post-install.in (hunk 3 above)
Comment (jmrojas2332) [reply oam/install_scripts/columnstore-post-install.in:320]: https://mdbcdt.com/DOCS-2085/deploy/community-single-columnstore/#post-installation

**Comment by jmrojas2332** [comment]
Closing to submit a cleaned up PR

**Pull request closed without merging**